### PR TITLE
CIV-5377 - Legal Rep ITP - Response Document and Mediation Screen.

### PR DESCRIPTION
**Pull request by neeta-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-5377

### Change description ###
Added a change to display response document screen when the claimant LR has rejected the part payment amount.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by KyleMulvihillV1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 1]
```diff
@@ -130,14 +131,42 @@ private CallbackResponse validateApplicantExperts(CallbackParams callbackParams)
...
             updatedCaseData.applicant1ProceedWithClaim(YES);
         }
+        return updatedCaseData.build();
+    }
+
+    private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
+        var updatedCaseData = caseData.toBuilder();
+
+        if (NO.equals(caseData.getApplicant1AcceptAdmitAmountPaidSpec())
+            || (caseData.getRespondent1ClaimResponseTypeForSpec().equals(RespondentResponseTypeSpec.FULL_DEFENCE)
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:155]: i would still put this in a boolean method that is more descriptive of what the condition is you are testing. sorry to be picky

Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:155]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 2]
```diff
@@ -130,14 +131,49 @@ private CallbackResponse validateApplicantExperts(CallbackParams callbackParams)
     }
 
     private CallbackResponse setApplicant1ProceedFlag(CallbackParams callbackParams) {
-        CaseData caseData = callbackParams.getCaseData();
+        CaseData updatedCaseData = setApplicant1ProceedFlagToYes(callbackParams.getCaseData());
```
Comment (kenneth-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:134]: Can you try to create a 2v1 case (LR,LR v LR) with same response to see what happens after your change of the code

**Review by TatsyMaslova — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 2 above)
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:134]: tested.. working as expected.

**Review by TatsyMaslova — APPROVED** [review]

**Review by kenneth-hmcts — DISMISSED** [review]

**Review by TatsyMaslova — DISMISSED** [review]

**Pull request closed without merging**